packages/rpc-core/src/rpc-methods/__tests__/get-account-info-test.ts

```diff
@@ -0,0 +1,182 @@
+
+            const accountInfo = await rpc
+                .getAccountInfo(publicKey, {
+                    encoding: 'base58',
+                })
+                .send();
+
+            expect(accountInfo.value?.data).toStrictEqual(['2Uw1bpnsXxu3e', 'base58']);
+        });
+    });
+
+    describe('when called with base64 encoding', () => {
+        it('returns account info with annotated base64 encoding', async () => {
+            expect.assertions(1);
+            // See scripts/fixtures/GQE2yjns7SKKuMc89tveBDpzYHwXfeuB2PGAbGaPWc6G.json
+            // data is 'test data'
+            const publicKey =
+                'GQE2yjns7SKKuMc89tveBDpzYHwXfeuB2PGAbGaPWc6G' as Base58EncodedAddress<'GQE2yjns7SKKuMc89tveBDpzYHwXfeuB2PGAbGaPWc6G'>;
+
+            const accountInfo = await rpc
+                .getAccountInfo(publicKey, {
+                    encoding: 'base64',
+                })
+                .send();
+
+            expect(accountInfo.value?.data).toStrictEqual(['dGVzdCBkYXRh', 'base64']);
+        });
+    });
+
+    describe('when called with base64+zstd encoding', () => {
+        it('returns account info with annotated base64+zstd encoding', async () => {
+            expect.assertions(1);
+            // See scripts/fixtures/GQE2yjns7SKKuMc89tveBDpzYHwXfeuB2PGAbGaPWc6G.json
+            // data is 'test data'
+            const publicKey =
+                'GQE2yjns7SKKuMc89tveBDpzYHwXfeuB2PGAbGaPWc6G' as Base58EncodedAddress<'GQE2yjns7SKKuMc89tveBDpzYHwXfeuB2PGAbGaPWc6G'>;
+
+            const accountInfo = await rpc
+                .getAccountInfo(publicKey, {
+                    encoding: 'base64+zstd',
+                })
+                .send();
+
+            expect(accountInfo.value?.data).toStrictEqual(['KLUv/QBYSQAAdGVzdCBkYXRh', 'base64+zstd']);
+        });
+    });
+
+    describe('when called with jsonParsed encoding', () => {
+        describe('for an account without parse-able JSON data', () => {
+            it('falls back to annotated base64', async () => {
+                expect.assertions(1);
+                // See scripts/fixtures/GQE2yjns7SKKuMc89tveBDpzYHwXfeuB2PGAbGaPWc6G.json
+                const publicKey =
+                    'GQE2yjns7SKKuMc89tveBDpzYHwXfeuB2PGAbGaPWc6G' as Base58EncodedAddress<'GQE2yjns7SKKuMc89tveBDpzYHwXfeuB2PGAbGaPWc6G'>;
+
+                const accountInfo = await rpc
+                    .getAccountInfo(publicKey, {
+                        encoding: 'jsonParsed',
+                    })
+                    .send();
+
+                expect(accountInfo.value?.data).toStrictEqual(['dGVzdCBkYXRh', 'base64']);
+            });
+        });
+
+        describe('for an account with parse-able JSON data', () => {
+            // TODO: not sure how to do this in the test validator
+            // Creating an account with data from a mainnet stake account doesn't work
+            // I think the testnet validator might not have the parsers?
```

Comment:
I just tried this and it worked.

1. Find an account that definitely parses on mainnet (eg. `CSg2vQGbnwWdSyJpwK4i3qGfB6FebaV3xQTx4U1MbixN`)
2. `solana-test-validator -u m --clone CSg2vQGbnwWdSyJpwK4i3qGfB6FebaV3xQTx4U1MbixN`

```
curl http://localhost:8899 -X POST -H "Content-Type: application/json" -d '
  {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "getAccountInfo",
    "params": [
      "CSg2vQGbnwWdSyJpwK4i3qGfB6FebaV3xQTx4U1MbixN",
      { "encoding": "jsonParsed" }
    ]
  }
'
```

Reply:
Thanks for the pointer! Figured it out, I was using the wrong `owner` in the fixture and I guess it uses that to figure out how to parse the account. Makes sense!